bokeh/command/subcommands/serve.py

```diff
@@ -917,8 +917,14 @@ def show_callback() -> None:
             if server.address is not None and server.address != '':
                 address_string = server.address
 
+            if server_kwargs['ssl_certfile'] and (
+                    server_kwargs['ssl_certfile'].endswith('.pem') or server_kwargs['ssl_keyfile']):
+                protocol = 'https'
+            else:
+                protocol = 'http'
+
             for route in sorted(applications.keys()):
-                url = "http://%s:%d%s%s" % (address_string, server.port, server.prefix, route)
+                url = "%s://%s:%d%s%s" % (protocol, address_string, server.port, server.prefix, route)
```

Comment:
```suggestion
                url = f"{protocol}://{address_string}:{server.port}{server.prefix}{route}"
```
@TheoMathurin as long as we are updating this let's update to use an f-string as well

Reply:
Yes ok, I thought I would not meddle with the choice of string formatting, some people have their reasons to keep using the % operator.

Just checked on my config and it works the same.

Reply:
The only reason I can think of in our codebase is to defer evaluation of arguments to logging, but we are constructing the string up front already anyway, so that's not really a concern here!